modules/extensions-spring-support/src/test/resources/models/subtypes.json

```diff
@@ -49,7 +49,25 @@
     {
       "baseType": {
         "format": "java",
-        "type": "@ref:org.mule.test.heisenberg.extension.model.Weapon"
+        "type": "Object",
+        "annotations": {
+          "typeId": "org.mule.test.heisenberg.extension.model.Weapon",
```

Comment:
This happened because we are no longer adding the imported types to the extension's type catalog, so the type is defined multiple times (in the original extension's catalog and all the places where it's used in the other extensions that import the same type)